lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb

```diff
@@ -2,15 +2,39 @@ module Fastlane
   module Actions
     class IosGenerateStringsFileFromCodeAction < Action
       def self.run(params)
-        files = files_matching(paths: params[:paths], exclude: params[:exclude])
-        flags = [('-q' if params[:quiet]), ('-SwiftUI' if params[:swiftui])].compact
-        flags += Array(params[:routines]).flat_map { |routine| ['-s', routine] }
-        cmd = ['genstrings', '-o', params[:output_dir], *flags, *files]
-        out = Actions.sh_control_output(*cmd, print_command: FastlaneCore::Globals.verbose?, print_command_output: true)
-        out = out.scrub.strip.split("\n")
-        errors = out.select { |line| line.include?('genstrings: error: ') }
-        UI.user_error!(errors.join("\n")) unless !params[:fail_on_error] || errors.empty?
-        out
+        output_encoding = begin
+          Encoding.find(params[:output_encoding])
+        rescue ArgumentError => e
+          UI.user_error!(e.message)
+        end
+
+        Dir.mktmpdir('genstrings-output-') do |tmpdir|
+          # Build the command arguments
+          files = files_matching(paths: params[:paths], exclude: params[:exclude])
+          flags = [
+            ('-q' if params[:quiet]),
+            ('-SwiftUI' if params[:swiftui]),
+            # If no endianness (-bigEndian vs -littleEndian) is specified, genstrings will use endianness of the current OS.
+            # Currently, genstrings runs only on macOS, which is little-endian, so this parameter is not strictly necessary.
+            # We make it explicit here to raise visibility on the relationship between the endianness of the genstring output and that of the encoding later on.
+            '-littleEndian'
```

Comment:
<table data-meta="generated_by_pr-check">
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/wordpress-mobile/release-toolkit/blob/4d955f19cd2276604d2478dfab63b06c44c0bf94/lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_generate_strings_file_from_code.rb#L20"></span>Style/TrailingCommaInArrayLiteral: Put a comma after the last item of a multiline array.</td>
    </tr>
  </tbody>
</table>

Reply:
```suggestion
            '-littleEndian',
```